### PR TITLE
Add column 'recordTimestamp' to record table

### PR DESCRIPTION
**Pull request by ghost**
Add column 'recordTimestamp' to record table. The value is parsed from the 005-field.

### CODE REVIEW

**Comment by ghost** [comment]
Depends on NatLibFi/melinda-deduplication-common#1

File: src/datastore-service.js [hunk 1]
```diff
@@ -179,7 +179,7 @@ function createDataStoreService(connectionPool: any): DataStoreService {
       record: record.toString(),
       parentId: RecordUtils.parseParentId(record),
       timestamp: now,
-      recordTimestamp: moment(record.get(/^005$/).shift().value, 'YYYYMMDDHHmmssSS').format('YYYY-MM-DDTHH:mm:ss.SS')
+      recordTimestamp: moment(_.chain(record.get(/^005$/)).shift().value, 'YYYYMMDDHHmmssSS').format('YYYY-MM-DDTHH:mm:ss.SS')
```
Comment (ghost) [inline src/datastore-service.js]: Without `_.chain` Flow reports and error:
```sh
property `value`. Property not found in
```

I don't know if there's a better way to solve this problem (Flow type definition is a union consisting of ControlField and DataField).

File: src/datastore-service.js [hunk 2]
```diff
@@ -177,22 +178,25 @@ function createDataStoreService(connectionPool: any): DataStoreService {
       base: base,
       record: record.toString(),
       parentId: RecordUtils.parseParentId(record),
-      timestamp: now
+      timestamp: now,
+      recordTimestamp: moment(_.chain(record.get(/^005$/)).shift().value, 'YYYYMMDDHHmmssSS').format('YYYY-MM-DDTHH:mm:ss.SS')
```
Comment (petuomin) [inline src/datastore-service.js]: This logic should be in RecordUtils. RecordUtils.getRecordTimestamp() or something like that.

File: src/schema/datastore-schema.js [hunk 3]
```diff
@@ -15,6 +15,7 @@ create table if not exists record (
   record text,
   parentId varchar(60),
   timestamp BIGINT NOT NULL,
+  recordTimestamp DATETIME NOT NULL,
```
Comment (petuomin) [inline src/schema/datastore-schema.js]: The initial schema should not be changed. The schema should only be changed with migrations to ensure that the migrations have a known previous database state to migrate from.

File: src/schema/migrations/migration-5-to-6.sql [hunk 4]
```diff
@@ -0,0 +1,4 @@
+set @record_timestamp_index_exists = (SELECT COUNT(1) IndexIsThere FROM INFORMATION_SCHEMA.STATISTICS WHERE table_schema=DATABASE() AND table_name='record' AND index_name='record_timestamp');
```
Comment (petuomin) [inline src/schema/migrations/migration-5-to-6.sql:3]: Alter table record to add a column for the record timestamp before adding index.

**Review by petuomin — CHANGES_REQUESTED** [review]

**Pull request merged**